dbt/clients/jinja.py

```diff
@@ -58,10 +58,39 @@ def _compile(self, source, filename):
         return super(MacroFuzzEnvironment, self)._compile(source, filename)
 
 
-def macro_generator(template, node):
+class TemplateCache(object):
+
+    def __init__(self):
+        self.file_cache = {}
+
+    def get_node_template(self, node):
+        # multiple nodes can be in a file, but they must have unique names!
+        key = (node['original_file_path'], node['name'])
```

Comment:
the key should be unique-per-file, instead of unique-per-node, no?

Reply:
I'm pretty confident unique-per-node is correct!
 - models are a single node per file and their `raw_sql` is the full file, so it's exactly the same
 - macros/operations have multiple nodes per file, and the `raw_sql` is the block contents, and each block has a (project-unique, which is vastly more than we care about) unique name

I also considered changing the key to the node's `raw_sql` field, but that seemed unreasonably large.

One thing we might need to do is add in the root path/project name to the key. I hadn't considered dependent projects that happen to have an identical layout until just now. It seems that at least in some tests, `original_file_path` is actually the just the path under `macros/`.

Reply:
Ok, you were totally correct. It's all about dependency packages, I misunderstood the issue because I missed the difference between `do_something` and `do_something2`!